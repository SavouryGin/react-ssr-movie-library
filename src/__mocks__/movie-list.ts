export const movieList = [
  {
    title: 'Pulp Fiction',
    genres: ['Action', 'Adventure'],
    year: 2004,
    imagePath: '__mocks__/images/Bitmap-1.png',
  },
];
