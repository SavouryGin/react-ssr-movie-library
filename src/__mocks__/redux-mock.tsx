import { combineReducers } from '@reduxjs/toolkit';
import { moviesInitialState, moviesSlice } from 'store/movies/slice';

export const mockedReducer = combineReducers({
  movies: moviesSlice.reducer,
});

export const mockedState = {
  movies: moviesInitialState,
};
