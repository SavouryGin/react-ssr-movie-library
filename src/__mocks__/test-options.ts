import { SortingValues } from 'enums/params';
import { MultiSelectOption, SelectEntity } from 'types/controls';

export const mockSortOptions: SelectEntity[] = [
  {
    option: 'Test-1',
    value: 1,
  },
  {
    option: 'Test-2',
    value: 2,
  },
  {
    option: 'Test-3',
    value: 3,
  },
];

export const mockMultiSelectOptions: MultiSelectOption[] = [
  {
    label: 'Test-1',
    value: 1,
  },
  {
    label: 'Test-2',
    value: 2,
  },
  {
    label: 'Test-3',
    value: 3,
  },
];
