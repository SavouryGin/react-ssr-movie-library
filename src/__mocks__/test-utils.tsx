import React, { ReactElement, ReactNode } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Reducer, configureStore } from '@reduxjs/toolkit';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { Field, Form } from 'react-final-form';

export const renderWithRedux = (ui: ReactElement, reducer: Reducer, preloadedState = {}, renderOptions?: RenderOptions): RenderResult => {
  const mockedStore = configureStore({ reducer, preloadedState });
  const Wrapper: React.FC = ({ children }: { children?: ReactNode }) => {
    return (
      <Provider store={mockedStore}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  };

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export const renderInsideForm = (ui: ReactElement, renderOptions?: RenderOptions): RenderResult => {
  const Wrapper: React.FC = ({ children }: { children?: ReactNode }) => {
    return (
      <Form
        onSubmit={jest.fn()}
        render={() => {
          return <>{children}</>;
        }}
      />
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};
