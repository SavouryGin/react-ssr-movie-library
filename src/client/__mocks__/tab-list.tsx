import React from 'react';
import { TabItem } from 'types/tabs';

export const testTabs: TabItem[] = [
  { tabContent: <>Tab content 1</>, tabId: 'tab-1', tabTitle: 'Title 1' },
  { tabContent: <>Tab content 2</>, tabId: 'tab-2', tabTitle: 'Title 2' },
  { tabContent: <>Tab content 3</>, tabId: 'tab-3', tabTitle: 'Title 3' },
  { tabContent: <>Tab content 4</>, tabId: 'tab-4', tabTitle: 'Title 4' },
];
