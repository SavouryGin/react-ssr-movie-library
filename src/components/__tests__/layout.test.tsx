import Layout from 'components/layout/index';
import React from 'react';
import { mockedReducer, mockedState } from '__mocks__/redux-mock';
import { renderWithRedux } from '__mocks__/test-utils';
import { screen } from '@testing-library/react';

describe('Layout component:', () => {
  beforeEach(() => {
    renderWithRedux(<Layout />, mockedReducer, mockedState);
  });

  it('matches the snapshot', () => {
    const { asFragment } = renderWithRedux(<Layout />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the header component', () => {
    expect(screen.getByRole('banner')).toBeInTheDocument();
  });

  it('renders the footer component', () => {
    expect(screen.getByRole('contentinfo')).toBeInTheDocument();
  });

  it('renders the main component', () => {
    expect(screen.getByTestId('main-container')).toBeInTheDocument();
  });
});
