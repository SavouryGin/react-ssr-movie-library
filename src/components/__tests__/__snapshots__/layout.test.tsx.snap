// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component: matches the snapshot 1`] = `
<DocumentFragment>
  <div
    data-cy="page-layout"
  >
    <header
      class=""
    >
      <div>
        <span>
          <strong
            class="icon_calendar"
          >
            netflix
          </strong>
          roulette
        </span>
        <button
          class=""
          title="Add movie"
          type="button"
        >
          <span
            class=""
          >
            Add movie
          </span>
        </button>
      </div>
      <div />
    </header>
    <main
      data-testid="main-container"
    />
    <footer
      class=""
    >
      <span>
        <strong>
          netflix
        </strong>
        roulette
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
